src/common/contracts.py

```diff
@@ -25,6 +25,10 @@
 class ContractWrapper:
     abi_path: str = ''
     settings_key: str = ''
+    execution_client: AsyncWeb3
```

Comment:
No need to annotate, because mypy will recognize `self.execution_client` from constructor.